src/components/Main/Main.test.js

```diff
@@ -0,0 +1,74 @@
+import { toBeInTheDocument } from "@testing-library/jest-dom/dist/matchers";
+import {
+  findByAltText,
+  getByAltText,
+  getByText,
+  render,
+  screen,
+  waitFor,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import moment from "moment";
+import Main from "./Main";
+
+describe("check components should render", () => {
+  test("check the datepicker exists", () => {
+    render(<Main />);
+    const today = moment().format("YYYY-MM-DD");
+
+    const textElement = screen.getByDisplayValue(today);
+    expect(textElement).toBeInTheDocument();
+  });
+
+  test("show loading", () => {
+    render(<Main />);
+    const today = moment().format("YYYY-MM-DD");
+    const dayAfterToday = moment().format("D");
+    const dateElement = screen.getByDisplayValue(today);
+
+    userEvent.click(dateElement);
+    const elementFromCalendar = screen.getAllByText(dayAfterToday)[0];
+    expect(elementFromCalendar).toBeInTheDocument();
+    userEvent.click(elementFromCalendar);
+    expect(screen.getByText(/loading/i)).toBeInTheDocument();
+  });
+});
+
+describe("Select a date", () => {
+  test("select a wrong date (later) and display it error message", async () => {
+    render(<Main />);
+    const today = moment().format("YYYY-MM-DD");
+    const dayAfterToday = moment().add(1, "days").format("D");
+    const todayFormat = moment().format("MMM DD, YYYY");
+    const dateElement = screen.getByDisplayValue(today);
+
+    userEvent.click(dateElement);
+    const elementFromCalendar = screen.getAllByText(dayAfterToday)[0];
+    expect(elementFromCalendar).toBeInTheDocument();
+    userEvent.click(elementFromCalendar);
+    await waitFor(() => {
+      expect(
+        screen.getByText(new RegExp(todayFormat, "i"))
+      ).toBeInTheDocument();
+    });
+  });
+
```

Comment:
Since datepicker is implemented, there are several ways to trigger acceptance criteria 3, which would allow your app to display a message to the user in case something unexpected happens, like network errors, api is down, params wrongly passed or the url is no longer valid.

```suggestion
  test("When the app fetches the API, and there is an unexpected error, the app should show a message: 'There was an error, please try again.'", async () => {
    process.env.REACT_APP_API_URL = "wrong_url";
    await render(<Main />);
    const expectedErrorMessage = await waitFor(() =>
      screen.getAllByText(/there was an error, please try again/i)
    );
    expect(expectedErrorMessage[0]).toBeInTheDocument();
  });
```